pow/src/test/java/tech/pegasys/teku/pow/FallbackAwareEth1ProviderSelectorTest.java

```diff
@@ -0,0 +1,217 @@
+    final List<Eth1Provider> providers = Arrays.asList(node1, node2, node3);
+    final Eth1ProviderSelector providerSelector = new Eth1ProviderSelector(providers);
+    final FallbackAwareEth1Provider fallbackAwareEth1Provider =
+        new FallbackAwareEth1Provider(providerSelector, asyncRunner);
+    // node 1 ready
+    when(node1.getLatestEth1Block()).thenReturn(readyProviderEthBlock());
+    fallbackAwareEth1Provider.getLatestEth1Block();
+
+    // node 1 failing and node 2 ready
+    when(node1.getLatestEth1Block()).thenReturn(failingProviderEthBlock());
+    when(node2.getLatestEth1Block()).thenReturn(readyProviderEthBlock());
+    final SafeFuture<EthBlock.Block> latestEth1Block =
+        fallbackAwareEth1Provider.getLatestEth1Block();
+    assertThat(latestEth1Block.get()).isNotNull();
+
+    // node 1 failing and node 2 are failing
+    when(node1.getLatestEth1Block()).thenReturn(failingProviderEthBlock());
+    when(node2.getLatestEth1Block()).thenReturn(failingProviderEthBlock());
+    assertThat(fallbackAwareEth1Provider.getLatestEth1Block()).isCompletedExceptionally();
+
+    verify(node1, times(3)).getLatestEth1Block();
+    verify(node2, times(2)).getLatestEth1Block();
+  }
+
+  @Test
+  void shouldFallbackWhileHonoringGuarantees() throws ExecutionException, InterruptedException {
+    final StubAsyncRunner asyncRunner = new StubAsyncRunner();
+    final Eth1Provider node1 = mock(Eth1Provider.class);
+    final Eth1Provider node2 = mock(Eth1Provider.class);
+    final Eth1Provider node3 = mock(Eth1Provider.class);
+    final List<Eth1Provider> providers = Arrays.asList(node1, node2, node3);
+    final Eth1ProviderSelector providerSelector = new Eth1ProviderSelector(providers);
+    final FallbackAwareEth1Provider fallbackAwareEth1Provider =
+        new FallbackAwareEth1Provider(providerSelector, asyncRunner);
+
+    /** * blockNumber ** */
+
+    // node 1,2 and 3 are all failing
+    when(node1.getEth1Block((UInt64) any())).thenReturn(failingProviderOptionalEthBlock());
+    when(node2.getEth1Block((UInt64) any())).thenReturn(failingProviderOptionalEthBlock());
+    when(node3.getEth1Block((UInt64) any())).thenReturn(failingProviderOptionalEthBlock());
+    final SafeFuture<EthBlock.Block> blockByNumber =
+        fallbackAwareEth1Provider.getGuaranteedEth1Block((UInt64) any());
+
+    // we expect a delayed action after all nodes has been contacted
+    assertThat(asyncRunner.hasDelayedActions()).isTrue();
+    verify(node1, atLeastOnce()).getEth1Block((UInt64) any());
+    verify(node2, atLeastOnce()).getEth1Block((UInt64) any());
+    verify(node3, atLeastOnce()).getEth1Block((UInt64) any());
+
+    // now we set node2 to be responsive and execute queued actions
+    when(node2.getEth1Block((UInt64) any())).thenReturn(readyProviderOptionalEthBlock());
+    asyncRunner.executeQueuedActions();
+
+    // we should have now a block
+    assertThat(blockByNumber.get()).isNotNull();
```

Comment:
It would probably be good to split out different tests for each method we're testing rather than having them all combined in one test.  Could likely pull a lot of the mocks out to fields so they can be reused between tests etc to avoid too much duplication.